lib/api/controllers/adminController.js

```diff
@@ -92,20 +89,24 @@ class AdminController extends BaseController {
       database = request.input.args.database,
       cacheEngine = this.kuzzle.services.list[database];
 
-    if (! (cacheEngine !== undefined && typeof cacheEngine.flushdb === 'function')) {
-      throw new BadRequestError(`Database ${database} not found`);
+    if (! (cacheEngine !== undefined &&
+        typeof cacheEngine.flushdb === 'function')) {
+      this.throw('database_not_found', database);
     }
 
     cacheEngine.flushdb();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') TypeError can be thrown as "cacheEngine" might be null or undefined here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS2259)